definitions/npm/react-apollo_v2.x.x/flow_v0.58.x-/react-apollo_v2.x.x.js

```diff
@@ -276,25 +276,38 @@ declare module 'react-apollo' {
 
   declare export class Subscription<TData> extends React$Component<SubscriptionProps<TData>> {}
 
-  declare export type MutationFunction<TVariables> = (options: {
+  declare type ExecutionResult<T> = {
+    data?: T;
+    extensions?: {[string]: any};
+    errors?: any[];
+  }
+
+  declare type FetchResult<C={[string]: any}, E={[string]: any}> = ExecutionResult<C> & {
+    extensions?: E,
+    context?: C
+  }
+
+  declare type OperationVariables = {[string]: any}
+
+  declare export type MutationFunction<TData=any, TVariables=OperationVariables> = (options: {
     variables?: TVariables,
     optimisticResponse?: Object,
-    refetchQueries?: (mutationResult: FetchResult) => Array<{query: DocumentNode, variables: {[string]: any}}>,
-    update?: (cache: DataProxy, mutationResult: FetchResult) => any
-  }) => Promise<*>
+    refetchQueries?: (mutationResult: FetchResult<>) => Array<{query: DocumentNode, variables: {[string]: any}}>,
```

Comment:
This should have the same type as refetchQueries below https://github.com/apollographql/react-apollo/blob/master/src/types.ts#L27